src/Response.php

```diff
@@ -85,9 +85,24 @@ private function getUpdateVersion() : array {
 			return [];
 		}
 
-		$values = $this->config[$this->oem][$this->platform];
-		if(version_compare($this->version, $values['version']) === -1) {
-			return $values;
+		// throttle down: starting with 10% increase every 12h by 10% -> after 5d 100%
+		$releaseTimeStamp = date_create($this->config[$this->oem]['release']);
+
+		if (isset($releaseTimeStamp)) {
+			$elapsedHours = (int) date_diff(date_create(), $releaseTimeStamp)->format("%h");
+			$throttle = intdiv($elapsedHours, 10);
+
+			if ($throttle >= rand(0, 10)) {
```

Comment:
random is bad, because it means after receiving a success message on the next request the client could get the old version again and back and forth between that all the time.

Reply:
Let's just store the "current time" in the nextcloud.cfg and append that to the request url. should be good enough like it is for the server.

Reply:
True. But this means that it will work only after next update, and not now.
Hopefully Michael and me can figure out something for a quick win.

Reply:
Using the request IP address could a bit more stable, at least for a couple of hours

Reply:
Could be, but as this is only for transition, I would leave it as simple as possible.